### ISSUE TITLE
Left justify the circuit in text visualizer

### ISSUE THREAD

**Issue by maddy-tod** [Qiskit/qiskit-terra#1707]
### What is the expected enhancement?
Currently when a circuit is displayed using the text drawer each gate in a circuit sits in its own column, as shown in the following circuit.
``` 
         ┌───┐          
q1_0: |0>┤ H ├──────────
         └───┘┌───┐     
q1_1: |0>─────┤ H ├─────
              └───┘     
```
It would be better if the circuits could be left justified, so the above circuit would appear as below.
```
         ┌───┐          
q1_0: |0>┤ H ├───────
         └───┘
         ┌───┐          
q1_1: |0>┤ H ├───────
         └───┘
```
It appears that circuits are already justified like this in the matplotlib drawer.

**Comment by maddy-tod** [comment]
In a circuit such as
```
circuit.h(qr1[1])
circuit.cx(qr1[0], qr1[3])
circuit.h(qr1[2])
```
The fully left justified circuit would be
```                   
q1_0: |0>──■───────
           │  ┌───┐
q1_1: |0>──┼──┤ H ├
           │  ├───┤
q1_2: |0>──┼──┤ H ├
         ┌─┴─┐└───┘
q1_3: |0>┤ X ├─────
         └───┘    
```
However this now no longer respects the order in which the gates were declared, and I think would be less helpful to users. I think it would be better to follow the style already used in the matplotlib drawer and produce a circuit where the gates are in their own columns, in the order they were declared. This will take up more space but would be clearer.

**Comment by ajavadia** [comment]
This is what matplotlib does. Do you mean this?
![image](https://user-images.githubusercontent.com/8622381/51628012-ddf16280-1f11-11e9-959e-be94fb224c4c.png)

Or do you mean this?
```
                        
q3_0: |0>───────■───────
         ┌───┐  │       
q3_1: |0>┤ H ├──┼───────
         └───┘  │  ┌───┐
q3_2: |0>───────┼──┤ H ├
              ┌─┴─┐└───┘
q3_3: |0>─────┤ X ├─────
              └───┘     
```

You are right that the second one is more accurate in terms of gate order, but then that would not really be left justified. The reason I was proposing left justified is that it gives you a good sense of the circuit depth. Except when it is not visually appealing (in this example, circuit depth is 1, but you can't really draw it with a depth of 1, so you add one more column).

**Comment by maddy-tod** [comment]
When I output that circuit to matplotlib the image I get is
![image](https://user-images.githubusercontent.com/40489777/51668605-7fcc8a00-1fba-11e9-84ff-238deefc0adb.png)
which is the same as your second example. I get a different image because I was running it against a version of qiskit that includes my fix for #1685 - sorry for the confusion! It might make sense to have the circuits right justified if this is what is done on the devices, and this would also give a sense of the circuit depth.

**Comment by ajavadia** [comment]
Sure, I think there is merit to a right justified drawing as well. Left-justified is common in backend-agnostic circuits (textbooks, papers). Also the device schedule is something that is backend-specific and can change with different schedulers.

I think if we could have this as a drawing option in `circuit.draw()` it would be best (maybe `justify='left'/'right'` or `schedule='asap'/'alap'`).
